metadata/src/main/java/org/apache/kafka/controller/AclControlManager.java

```diff
@@ -152,7 +154,11 @@ ControllerResult<List<AclDeleteResult>> deleteAcls(List<AclBindingFilter> filter
                 results.add(new AclDeleteResult(ApiError.fromThrowable(e).exception()));
             }
         }
-        return ControllerResult.atomicOf(records, results);
+        return ControllerResult.atomicOf(dedupeApiMessageAndVersion(records), results);
+    }
+
+    private List<ApiMessageAndVersion> dedupeApiMessageAndVersion(List<ApiMessageAndVersion> messages) {
+        return new HashSet<>(messages).stream().collect(Collectors.toList());
```

Comment:
nit: doesn't make a big difference, but any reason not to begin with a Set instead of converting from a List?

Reply:
Good point. I think we could use a Set the whole way through except at the very end when we have to convert it to a List.

Originally I was thinking about preserving the order of the messages as I did the de-dupe. I had the following at first:
```
return new LinkedHashSet<>(messages).stream().collect(Collectors.toList());
```
but then I realized the ordering shouldnt matter. So I think I can make the suggested change to just use a Set.